footylib/footylib.py

```diff
@@ -56,85 +57,92 @@ def competitions(self):
-                    possible_teams.append(teams.name)
-        return "Found team(s): {}".format(possible_teams)
-
-    def get_team_season(self, team):
-        team_calendar = []
-        for competition in self.competitions:
-            for match in competition.matches:
-                if team in match.teams:
-                    team_calendar.append(dict(match=match.teams,
-                                              time=match.datetime))
-        return team_calendar
+            for team in competition.teams:
+                if team_name in team.name:
+                    possible_teams.append(team)
+        return possible_teams
 
 
 class Competition(object):
     def __init__(self, footy_instance, location, url, name):
-        self.logger = logging.getLogger('{base}.{suffix}'.format(
+        self._logger = logging.getLogger('{base}.{suffix}'.format(
             base=LOGGER_BASENAME, suffix=self.__class__.__name__))
         self.session = footy_instance.session
         self._populate(location, url, name)
         self._teams = []
         self._matches = []
+        self._calendar = None
 
     def _populate(self, location, url, name):
         try:
             self.location = location
             self.url = url
-            self.name = name.encode('utf-8')
+            self.name = name.encode('utf-8').strip()
         except KeyError:
-            self.logger.exception("Got an exception in Competition")
+            self._logger.exception("Got an exception in Competition")
 
     @property
     def teams(self):
         if not self._teams:
-            team_page = self.session.get(self.url)
-            soup = Bfs(team_page.text, "html.parser")
-            standings = soup.find_all('table',
-                                      {'class': 'leaguemanager standingstable'})
+            standings = self._get_table('standingstable')
             for teams in standings:
                 for row in teams.find_all('tr', {'class': ('alternate', '')}):
-                    self._teams.append(Team(row))
+                    self._teams.append(Team(self, row))
         return self._teams
 
     @property
     def matches(self):
         if not self._matches:
-            team_page = self.session.get(self.url)
-            soup = Bfs(team_page.text, "html.parser")
-            match_tables = soup.find_all('table',
-                                         {'class': 'leaguemanager matchtable'})
+            match_tables = self._get_table('matchtable')
             for match_table in match_tables:
                 division = match_table.attrs['title']
-                self._matches.extend([Match(row, division)
+                self._matches.extend([Match(self, row, division)
                                      for row in match_table.find_all('tr',
                                                                      {'class': ('alternate', '')})])
         return self._matches
 
+    def _get_table(self, class_attribute):
```

Comment:
You are not doing any caching of the competition page which is very expensive. Also lines 100-102 and 120-124 are actually the same but with a different syntax. They should probably be refactored into a single item.